server-common/src/main/java/org/apache/gravitino/server/authorization/GravitinoAuthorizerProvider.java

```diff
@@ -40,7 +41,21 @@ private GravitinoAuthorizerProvider() {}
    * @param serverConfig Gravitino server config
    */
   public void initialize(ServerConfig serverConfig) {
-    // TODO
+    if (gravitinoAuthorizer == null) {
+      synchronized (this) {
+        if (gravitinoAuthorizer == null) {
+          boolean enableAuthorization = serverConfig.get(Configs.ENABLE_AUTHORIZATION);
+          if (enableAuthorization) {
+            // TODO
+          } else {
+            gravitinoAuthorizer = new IgnoreAuthorizer();
+          }
+        }
+      }
+    }
+    if (gravitinoAuthorizer != null) {
```

Comment:
This should be protected by the lock, too. Otherwise, it may be initialize multiple times.